### DISCUSSION TITLE
Split on QR codes?

### DISCUSSION THREAD

**Discussion by tymanthius** [Stirling-Tools/Stirling-PDF#1726] [Ideas]
This may be too niche, but I have a client who needs to feed documents in and split them on QR codes.  The number of pages and even number of QR's will vary.  So maybe have the QR code name the split even?

Example:

Pge 1 is a barcode with file name - use that to create a top level folder with File Name and that page is never used again.
Page 2 is a QR code that says 'Buyer'
Pages 3-15 are for the Buyer and get split out into a file named Buyer in the File Name folder
Page 16 is a QR code that says 'Attny' 
Pages 17-8 are for the attny and get split out to a file named Attny in the File Name folder.

Proceed until all pages are done.

**Comment by Frooodle** [comment]
The top level stuff might be niche but general splitting by QR is common we do one already with our auto split but for Stirling pdf QR codes

There definitely are usecases for splitting by any QR code, I think it's a reasonable enhancement to want

Again not sure about that page 1 naming be rest seems good to add

**Reply by tymanthius** [reply]
That (top level file name) can be handled other ways too really.  

I did look and didn't see anything about QR code splitting - do you have a link?

**Reply by Frooodle** [reply]
https://stirlingpdf.io/auto-split-pdf

![image](https://github.com/user-attachments/assets/6b2f2386-2fda-4233-99eb-608dfe698446)


![image](https://github.com/user-attachments/assets/92bba53a-5924-4cbd-b294-7781160fcefe)

right now this doesnt split by any QR, only QRs containing a set value, but I think its a reasonable enhancement we can add for future

**Reply by tymanthius** [reply]
Ok, I got it. Thanks.  I'll see about playing w/ that, but yes, I think QR codes would add a HUGE amount of flexibility to this.  :)

**Comment by tymanthius** [comment]
Oh, should I add this as an official feature request?

**Reply by Frooodle** [reply]
please do!
